File: API/src/main/java/com/ltizzi/ecommerce/model/user/UserRequest.java
package com.ltizzi.ecommerce.model.user;

import com.ltizzi.ecommerce.model.cart.CartRequest;
import com.ltizzi.ecommerce.model.cart.CartResponse;
import com.ltizzi.ecommerce.model.purchase.PurchaseRequest;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * @author Leonardo Terlizzi
 */

@Data
@NoArgsConstructor
public class UserRequest {

    private Long user_id;
    private String username;
    private String name;
    private String lastname;
    private String email;
    private String avatar;
    private Date birthday;
    private List<CartRequest> carts = new ArrayList<>();
    private List<PurchaseRequest> purchases = new ArrayList<>();
}

File: Server/src/dto/requests/user.request.ts
import { Product } from "../../models/Product";
import { RoleEnum } from "../../models/RoleEnum";
import { CartRequest } from "./cart.request";
import { PurchaseRequest } from "./purchase.request";
import { Review } from "../../models/Review";

export interface UserRequest {
  user_id?: number;
  username: string;
  name: string;
  lastname: string;
  email: string;
  avatar: string;
  birthday: Date;
  carts?: Array<CartRequest>;
  purchases?: Array<PurchaseRequest>;
  roles: Array<RoleEnum>;
  favourites: Array<Product>;
  reviews: Array<Review>;
}
